docs/source/notes/designing_new_logic.rst

```diff
@@ -0,0 +1,351 @@
+                            say "You push the [pb], and [dr] is now open.";
+                        otherwise:
+                            say "You push the [pb] again, and [dr] is now locked."
+            """
+        }
+    }
+
+In this example, the "push" command is defined; the compiler expects to have a syntax such as "push [something]" which
+the [something] usually is replaced by the name assigned to the push button. It is also described that what changes
+are expected to happen when the button is pushed; the button state should change from `unpushed` to `pushed`, the door
+state also should be changed from `unlocked` to `open`. The last block is for human interaction and prints out these
+changes, thus, it is not mandatory.
+
+After defining the new instructions to model push button in the game based on inform7 language, next step is to define
+a command (or rule) to activate the action on both languages. "Rule" section is another section of logic file. It
+describes how the game transforms from one state to another by using the command; see the below example for a simple
+`open` rule for a door,
+
+.. code-block:: bash
+
+    open/d :: $at(P, r) & $link(r, d, r') & $link(r', d, r) & closed(d) -> open(d) & free(r, r') & free(r', r);
+
+where `at(P, r)` means "the player is at the room" and `free(r, r')` means the path from room r to room r' is clear.
+This rule includes two columns which are separated by a `::`. The left column presents the rule's name. This name
+should be unique for each rule, thus, if we have two states with different conditions, then their names should be
+different, for instance "open/d" vs "open/c" which stand for open door and open container, consecutively.
+
+The right column of the above rule describes the state change of the game according to the current change and the next
+state which the game will turn to, by using this command. As it is depicted, each state contains some predicates
+which describe the conditions applied to the elements of the game at that moment (or state) of the game, and provides
+eligibility for the defined rule to be applied/called. After calling the rule, it is activated and makes some changes
+into the state of the game (or equivalently some selected elements of the game) as it is described on the right side of
+the arrow. By these changes, the game will finally transit to the next state. Please be notified that any predicate
+which is supported by `$` sign will be kept as unchanged at the next state.
+
+Equivalently, the I7L version of this rule should be coded in the inform7 part, which is translated again as `open {d}`.
+When this command is imported by the player, the inform7 will return a response as the game state, which in this example
+is `opening {d}`. This inform-based output is important for the TextWorld framework to identify that the inform
+compiler has taken the action of the command (here opening the door) and has transited to the next state. This can be
```

Comment:
"that the inform compiler has taken the cation... " -> "that the game has understood the action..."